src/Illuminate/Collections/Arr.php

```diff
@@ -635,28 +635,14 @@ public static function random($array, $number = null, $preserveKeys = false)
         }
 
         if (is_null($number)) {
-            return $array[array_rand($array)];
+            return head(array_slice($array, random_int(0, $count - 1), 1));
         }
 
         if ((int) $number === 0) {
             return [];
         }
 
-        $keys = array_rand($array, $number);
-
-        $results = [];
-
-        if ($preserveKeys) {
-            foreach ((array) $keys as $key) {
-                $results[$key] = $array[$key];
-            }
-        } else {
-            foreach ((array) $keys as $key) {
-                $results[] = $array[$key];
-            }
-        }
-
-        return $results;
+        return array_slice(static::shuffle($array), 0, $number, $preserveKeys);
```

Comment:
Yeah, my bad. I added in preserving keys because I needed it for the `Arr:random($num)` usage, and overlooked the fact that shuffle didn't preserve them. I'll split them out so `random()` can preserve it's keys and `shuffle()` doesn't.